### PR TITLE
Mutable/immutable records doc changes and return value correction

### PR DESCRIPTION
**Pull request by 4c656554**

### CODE REVIEW

**Comment by mafintosh** [comment]
Can you set the branch to hyperswarm v3 ?

**Comment by 4c656554** [comment]
> Can you set the branch to hyperswarm v3 ?

ok

**Comment by 4c656554** [comment]
> > Can you set the branch to hyperswarm v3 ?
> 
> ok

Not seeing hyperswarm-v3 as a branch option for hyperswarm/dht (only on hyperswarm). Please do let me know where to fork from if it's easier to replace the PR, can easily re-edit files if necessary.

File: README.md [hunk 1]
```diff
@@ -31,35 +31,51 @@ Then on one computer listen for connections
...
+      // send a test message:
+      noiseSocket.write(Buffer.from('test'))
+    })
 
-// pipe it somewhere like any duplex stream
-process.stdin.pipe(noiseSocket).pipe(process.stdout)
+    // pipe it somewhere like any duplex stream
+    // process.stdin.pipe(noiseSocket).pipe(process.stdout) 
+
+}
```
Comment (mafintosh) [inline README.md:78]: can we remove the doc changes above or put that in a diff PR?

File: index.js [hunk 2]
```diff
@@ -311,7 +312,7 @@ module.exports = class HyperDHT extends DHT {
       }
     })
     await query.finished()
-    return { signature, seq }
+    return { closestNodes, signature, seq }
```
Comment (mafintosh) [inline index.js:315]: closestNodes: query.closestNodes

**Review by mafintosh — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (4c656554) [reply index.js]: hmm gives build error, is `query` from dht-rpc?
https://github.com/mafintosh/dht-rpc/blob/master/lib/query.js

Comment (mafintosh) [reply index.js]: Otherwise you can link the error here if you want soem help.

Comment (4c656554) [reply index.js]: This is the error:
https://github.com/hyperswarm/dht/pull/51/checks?check_run_id=3534495624

Comment (4c656554) [reply index.js]: `  D:\a\dht\dht\index.js:315:19: Parsing error: Unexpected token .
npm ERR! Test failed.  See above for more details.
Error: Process completed with exit code 1.`

Comment (4c656554) [reply index.js]: I think `query` is a `streamx` readable stream...

File: index.js [hunk 3]
```diff
@@ -311,7 +312,7 @@ module.exports = class HyperDHT extends DHT {
       }
     })
     await query.finished()
-    return { signature, seq }
+    return { query.closestNodes, signature, seq }
```
Comment (mafintosh) [inline index.js:315]:
```
closestNodes: query.closestNode
```

Comment (4c656554) [reply index.js:315]: This gives me:
`SyntaxError: Unexpected token '.'
[90m    at wrapSafe (internal/modules/cjs/loader.js:984:16)[39m
[90m    at Module._compile (internal/modules/cjs/loader.js:1032:27)[39m
[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)[39m
[90m    at Module.load (internal/modules/cjs/loader.js:933:32)[39m
[90m    at Function.Module._load (internal/modules/cjs/loader.js:774:14)[39m
[90m    at Module.require (internal/modules/cjs/loader.js:957:19)[39m
[90m    at require (internal/modules/cjs/helpers.js:88:18)[39m
    at Object.<anonymous> (D:\OODA_SPACE\oodaspacedev\hyperdhtoo\index2.js:1:13)
[90m    at Module._compile (internal/modules/cjs/loader.js:1068:30)[39m`

Comment (4c656554) [reply index.js:315]: logging `query` gives:
`mutablePut {
  query: <ref *1> Query {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    _duplexState: 2171022,
    _readableState: ReadableState {
      stream: [Circular *1],
      queue: [FastFIFO],
      highWaterMark: 0,
      buffered: 0,...`

Comment (4c656554) [reply index.js:315]: I think we may need to access before `query.finished()`

Comment (mafintosh) [reply index.js:315]: What exactly are you running? It should be:

```js
return { closestNodes: query.closestNodes, signature, seq }
```

Comment (4c656554) [reply index.js:315]: I'm unsure how to get data out of streamx objects...

Comment (4c656554) [reply index.js:315]: ok got it now! I should've read it properly!

Comment (mafintosh) [reply index.js:315]: It's a query object (that inherits from stream). It's defined using that getter above. If you are getting a syntax error it means there is a typo somewhere.

File: README.md (hunk 1 above)
Comment (mafintosh) [reply README.md:78]: Did you mean to resolve this? :)

Comment (4c656554) [reply README.md:78]: done

Comment (4c656554) [reply README.md:78]: yep, added a commit to revert back to original state of readme, is that what you had in mind?

Comment (4c656554) [reply README.md:78]: https://github.com/hyperswarm/dht/pull/51/commits/245bdae069b0dbfefe7f299bb1334ae486a58767

File: index.js (hunk 3 above)
Comment (4c656554) [reply index.js]: Was my bad, I was unfamiliar with getter in constructor syntax.

File: README.md (hunk 1 above)
Comment (mafintosh) [reply README.md:78]: It's still adding the README updates to your PR, you sure you added that to this branch?

**Pull request closed without merging**